### PR TITLE
Create a separate component for selecting organizations

### LINKED ISSUE #379: Filters - reuse and create components from common fields

### PR DESCRIPTION
**Pull request by gwynndp**
## Description

- Create a separate component for selecting organizations
- allow different handling of the organization by different filters
- Update filters to use the new component

**Issue(s) addressed**

- Resolves #379

**What kind of change(s) does this PR introduce?**

- [x] Enhancement
- [ ] Bug fix
- [ ] Refactor

**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Review by nmcharlton — COMMENTED** [review]
@gwynndp I'm very happy with the org selection refactor.
Are the `CustomTable` changes related?

**Comment by gwynndp** [comment]
> 

The only thing unrelated is that I broke out the Header from CustomTable into it's own component.  I can revert that if we don't want to break components out so we have one component per file.

How do we want to handle cleaning up code to match standards.  I often do it as I go but then it isn't related to the issue directly.  Do we want to create separate issues for cleanup?

**Comment by nmcharlton** [comment]
If the cleaning up is in the same place as the changes, it's fine to include them.
If it's general refactoring, it's best to split it into its own PR, so we can more easily identify where and why things have changed when there is a problem.

File: src/api/treeTrackerApi.js [hunk 1]
```diff
@@ -130,12 +131,15 @@ export default {
    * Capture Match Tool
    */
   fetchCapturesToMatch(currentPage, abortController, filter) {
-    const where = Object.keys(filter)
```
Comment (nmcharlton) [inline src/api/treeTrackerApi.js:133]: Was this not working correctly?

File: src/api/treeTrackerApi.js [hunk 2]
```diff
@@ -130,12 +131,15 @@ export default {
...
-      .map((key) => (filter[key] ? `${key}=${filter[key]}` : ''))
-      .join('&');
+    const where = Object.keys(filter).reduce(
+      (str, key) => (filter[key] ? (str += `&${key}=${filter[key]}`) : str),
+      ''
+    );
+
     const req = `${CAPTURE_MATCH_API}/captures?tree_associated=false&limit=${1}&offset=${
       currentPage - 1
     }&${where}`;
```
Comment (nmcharlton) [inline src/api/treeTrackerApi.js:141]: In the refactored `where` construction, there will always be a `&` at the beginning of the string already.

File: src/components/EarningsTable/EarningsTable.js [hunk 3]
```diff
@@ -123,28 +124,43 @@ function EarningsTable() {
...
   }
 
   async function getEarningsReal(fetchAll = false) {
-    console.warn('fetchAll:', fetchAll);
+    // console.warn('fetchAll:', fetchAll);
+    const filtersToSubmit = { ...filter };
+    // filter out keys we don't want to submit
+    Object.keys(filtersToSubmit).forEach((k) => {
+      if (
+        filtersToSubmit[k] === 'all' ||
```
Comment (nmcharlton) [inline src/components/EarningsTable/EarningsTable.js:143]: This is a risky way of spotting that all values should be included.
If a user wants to filter by a grower name that starts with or contains "all", we will ignore the value and return all growers.

Comment (nmcharlton) [reply src/components/EarningsTable/EarningsTable.js:143]: I see that it's just been moved from another part of the code, but we should try to resolve it.

**Review by nmcharlton — APPROVED** [review]
Looks good to me 👍

File: src/api/treeTrackerApi.js (hunk 1 above)
Comment (gwynndp) [reply src/api/treeTrackerApi.js:133]: I think I ended up with a "&&" at one point, but I'll have to review it again

File: src/api/treeTrackerApi.js (hunk 2 above)
Comment (gwynndp) [reply src/api/treeTrackerApi.js:141]: I'll check this solution again, thanks!

File: src/components/EarningsTable/EarningsTable.js (hunk 3 above)
Comment (gwynndp) [reply src/components/EarningsTable/EarningsTable.js:143]: I'll make another issue

**Comment by gwynndp** [comment]
@nmcharlton I'm not sure how to get the "npm clean install" step in the CI/CD to pass. Advice?

**Comment by Siddhant-K-code** [comment]
> @nmcharlton I'm not sure how to get the "npm clean install" step in the CI/CD to pass. Advice? It seems to be having trouble with `"react": "*"`, and `"react-dom": "*"` where the Material UI icons are concerned.

there is an inter-dependency issue with webpack versions from different packages.

we can use following command to ignore ^this error:
```yml
npm ci --legacy-peer-deps
```

**Review by Siddhant-K-code — APPROVED** [review]

**Comment by nmcharlton** [comment]
@gwynndp I've updated the scripts to use the flag suggested by @Siddhant-K-code. Can you pull the latest changes into your branch and make sure it builds ok?

**Comment by Siddhant-K-code** [comment]
Yeah, it should work now. No need to update branch, just make a empty commit & push it using: 

```
git commit --allow-empty -m "Empty commit"
```

```
git push
```

**Pull request merged**